scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -1011,7 +1011,14 @@ def report_out(self, results):
                     if instance.dut:
                         more_info += f": {instance.dut},"
                     if htime:
-                        more_info += " {:.3f}s".format(htime)
+                        hours = int(htime // 3600)
+                        minutes = int((htime % 3600) // 60)
+                        seconds = htime % 60
+                        if hours > 0:
+                            more_info += " {}h".format(hours)
+                        if minutes > 0 or hours > 0:
+                            more_info += " {}m".format(minutes)
+                        more_info += " {:.3f}s".format(seconds)
```

Comment:
We have datetime in Python (for both formatting and time calculation), let's not do it manually.

Reply:
I decided that such simple calculations do not require using additional libraries.
So. Twister gives us the time in float format. So I guess the only option to use this library is to use timedelta, e.g. `datetime.timedelta(seconds=htime)`
However, in this form we are not able to format it, e.g. using strftime(). That is why we can try to convert it to a string and then to a list.
`str(datetime.timedelta(seconds=htime)).split(':')`
However, this will not give the expected effect. In such a case, it will have to be formatted manually anyway. I do not want unnecessary zeros in the log. Most tests execute less than 60 seconds and this option is only for a few tests (but important). So if I still wanted to achieve a similar effect to what I have already written, I would have to complicates the code. Completely unnecessary. Like that:
```
time_form = str(datetime.timedelta(seconds=htime)).split(':')
if int(time_form[0]) > 0:
    more_info += " {}h".format(time_form[0])
if int(time_form[1]) > 0 or int(time_form[0]) > 0:
    more_info += " {}m".format(time_form[1])
more_info += " {:.3f}s".format(float(time_form[2]))
```

However, if you know a way to write it easily and concisely, I would like an example.

Reply:
> I do not want unnecessary zeros in the log. Most tests execute less than 60 seconds and this option is only for a few tests (but important).

```
more_info += " {:.3f}s".format(htime) if htime < 60 else " " + str(datetime.timedelta(seconds=htime))
```

Reply:
> > I do not want unnecessary zeros in the log. Most tests execute less than 60 seconds and this option is only for a few tests (but important).
> 
> ```
> more_info += " {:.3f}s".format(htime) if htime < 60 else " " + str(datetime.timedelta(seconds=htime))
> ```

After reformatting to make it look better, it still looks almost the same as I wrote. Because it would be nice if 00h wasn't displayed if the test lasts, for example, 84 s.
I still think it's an unnecessary complication...